pytorch/README.md

```diff
@@ -93,9 +93,9 @@ public class SimpleMNIST {
     static class Net extends Module {
         Net() {
             // Construct and register two Linear submodules.
-            fc1 = register_module("fc1", new LinearImpl(784, 64));
-            fc2 = register_module("fc2", new LinearImpl(64, 32));
-            fc3 = register_module("fc3", new LinearImpl(32, 10));
+            register_module("fc1", fc1 = new LinearImpl(784, 64));
+            register_module("fc2", fc2 = new LinearImpl(64, 32));
+            register_module("fc3", fc3 = new LinearImpl(32, 10));
```

Comment:
What's the reason why you removed those register_module() functions? If there is no particular reason why you removed them, please put them back.

Reply:
`register_module(String, Module)` was broken in 2.0.0 and earlier since:
```
Module m = new LinearImpl(32,10);
...
register_module("fc", m);
```
caused a segmentation fault.
This is why you added all the `register_module` overrides in the first place.
This was fixed in 2.0.1 with the `upcast` feature. So the overrides and the corresponding template instantiations were not necessary any more.
All what they do is returning a more specific type, but this is not useful in the general case since with a module `m` from a custom java class `JavaModule` extending `Module`,`register_module("x", m)` returns a `Module` and not the `JavaModule`.

Reply:
Right, but that doesn't justify breaking existing code. Put the old overrides back

Reply:
They were already removed in 2.0.1. And they alone increase the jni library size by about 4-5%. 
Can you confirm that's what you want before I push this commit ?

Reply:
Sure, that's fine by me. We could also add them back as helper Java methods like the ones here:
https://github.com/bytedeco/javacpp-presets/tree/master/opencv/src/main/java/org/bytedeco/opencv/helper

Reply:
Good idea. This looks better because it:
* doesn't instantiate tones of templates.
* allows to return the exact same object rather than a new wrapper against a C++ pointer
* will work with Java modules

Reply:
Something like:
```Java
static  <M extends Module> M register_module(String name, M module) { 
  _register_module(name, module);
return module;
}
```

Reply:
Oh and I think we can even use generics for that purpose